src/capabilities.js

```diff
@@ -0,0 +1,207 @@
+import { currentSchemaVersions } from '@mapeo/schema'
+import mapObject from 'map-obj'
+import { kCreateWithDocId } from './datatype/index.js'
+
+// Randomly generated 8-byte encoded as hex
+export const COORDINATOR_ROLE_ID = 'f7c150f5a3a9a855'
+export const MEMBER_ROLE_ID = '012fd2d431c0bf60'
+export const BLOCKED_ROLE_ID = '9e6d29263cba36c9'
+
+/**
+ * @typedef {object} DocCapability
+ * @property {boolean} readOwn - can read own data
+ * @property {boolean} writeOwn - can write own data
+ * @property {boolean} readOthers - can read other's data
+ * @property {boolean} writeOthers - can edit or delete other's data
+ */
+
+/**
+ * @typedef {object} Capability
+ * @property {string} name
+ * @property {Record<import('@mapeo/schema').MapeoDoc['schemaName'], DocCapability>} docs
+ * @property {RoleId[]} roleAssignment
+ * @property {'allowed' | 'blocked'} sync
+ */
+
+/**
+ * @typedef {typeof COORDINATOR_ROLE_ID | typeof MEMBER_ROLE_ID | typeof BLOCKED_ROLE_ID} RoleId
+ */
+
+/**
+ * This is currently the same as 'Coordinator' capabilities, but defined
+ * separately because the creator should always have ALL capabilities, but we
+ * could edit 'Coordinator' capabilities in the future
+ *
+ * @type {Capability}
+ */
+export const CREATOR_CAPABILITIES = {
+  name: 'Project Creator',
+  docs: mapObject(currentSchemaVersions, (key) => {
+    return [
+      key,
+      { readOwn: true, writeOwn: true, readOthers: true, writeOthers: true },
+    ]
+  }),
+  roleAssignment: [COORDINATOR_ROLE_ID, MEMBER_ROLE_ID, BLOCKED_ROLE_ID],
+  sync: 'allowed',
+}
+
+/** @type {Record<RoleId, Capability>} */
+export const DEFAULT_CAPABILITIES = {
+  [MEMBER_ROLE_ID]: {
+    name: 'Member',
+    docs: mapObject(currentSchemaVersions, (key) => {
+      return [
+        key,
+        { readOwn: true, writeOwn: true, readOthers: true, writeOthers: false },
+      ]
+    }),
```

Comment:
what's happening here with these object mappings? hard to tell at a glance without needing to dive into the `currentSchemaVersions` code